### ISSUE TITLE
Select isn't work all the time on MeshInstance and CSGBox in 3D space - Interface/UX Issue

### ISSUE THREAD

**Issue by thottee** [godotengine/godot#46037]
**Godot version:**
Tested with 3.1.x-3.2.x

**OS/device including version:**
Windows 8.1 - GTX1660Ti

**Issue description:**
You can't select with mouse click the MeshInstance Mesh (tested with just the CubeMesh) and the CSGBox in the 3D space at all the time, when you click on the mesh with the mouse, if you increase the mesh size at the mesh properties, or CSGBox extent properties. But you can select MeshInstance if you scale it with transfrom properties, but this thing don't work with CSGBox.

It's not a huge problem, but really annoying, when you want to just work on your project, and build a level from default meshes.

**Steps to reproduce:**
1. Create a MeshInstance with a CubeMesh
2. Set Mesh Size property to higher (I used: 40, 10 ,30)
3. Try to select this mesh with mouse (not on the center of mesh)

**Minimal reproduction project:**
I think it's easy to reproduce. I used godot default settings.

**Comment by Calinou** [comment]
This has been an issue for a long time and will require the object picking code to be reworked entirely for 4.0.

**Comment by timothyqiu** [comment]
I recently found dragging materials onto a mesh/CSG sometimes fails. Since material dragging & object picking shares the same code, I think they are related and this is probably a better way to reproduce the problem:

![Peek 2022-10-31 12-44](https://user-images.githubusercontent.com/372476/198932865-53d8d64f-68ec-4d40-81b3-a1aacc199186.gif)

As shown in the GIF, hovering on certain spots on the CSGPolygon picks the MeshInstance underneath.

**Comment by RetrocadeMedia** [comment]
As of 4.2 this is still a problem, and it seems to have gotten worse. I struggle to click on any csg mesh in my scene, and can not drag and drop a material on to them. I was able to get one random png to drop one time. In the video below, each time I move my mouse between nodes I am clicking, and they rarely get selected or select the correct one. You can also see I am unable to drag and drop a material.

https://youtu.be/dHKAcL24cCM

**Comment by notPelf** [comment]
This is still a problem on 4.3. I've dug a bit into reproducing the issue and here is what I've found...

I've noticed when I try clicking on a csg shape to select it, it seems to be immediately starting a box select. In these images, my cursor is hovering over a CSGBox3D. I have enlarged the images for clarity.

In the first image my cursor is hovering, no mouse buttons being clicked.
![Image](https://github.com/user-attachments/assets/fa614325-e9f1-453f-a0ab-9fa6f9d3063f)

In this next image, as soon as I click down, there is a single pixel which appears at the tip of the cursor....
![Image](https://github.com/user-attachments/assets/fe18f6d9-f1ee-4f21-a0ce-1dd5d05da841)

... which is the box select when click+dragging.
![Image](https://github.com/user-attachments/assets/b2da8e6e-b55f-4e36-bf54-8cf9095d4b81)

It seems when clicking on a CSG node, it _immediately_ starts a box select without waiting for the user to move the mouse. Normally a box select is only started once the mouse cursor moves a little after click+dragging.

However there is more to it...
1. Make a new 3D scene and place two CSGBox3D nodes in the scene as a child of the root node. Any size works.
![Image](https://github.com/user-attachments/assets/d6252d5b-9317-4610-b685-88b21b8a2cb3)
2. Click on the CSG boxes and notice that clicking and selecting works normally on them without any issue.
3. Now add a CSGCombiner3D node to the scene and make the boxes children of the combiner. Collision enabled/disabled doesn't matter.
![Image](https://github.com/user-attachments/assets/d87b8b66-2912-43dd-b031-e1d41ebb4b39)
4. Click on the boxes again and notice it tries doing the box select thing. However there is still more to it. There is a faint wireframe visible. If you click on the wireframe, selection works normally. If you do not click on the wireframe then it does the box select.

In summary, when using CSGCombiner3D nodes, the editor _immediately_ starts doing a box select without waiting for the mouse to move, but only if you do not click on the wireframe of the CSG mesh.

**Comment by XandaPanda42** [comment]
Still there on 4.4.1. 

Thanks to @notPelf for the wireframe tip. Saved me a lot of time in the last few weeks.

Using a CSGCombiner, with a with a CSGBox of size(8, 0.1, 8), with a black material for visibility. Switching to a top-down orthographic camera looking down on it showed some interesting stuff. 

1. Diagonal Bar.

Orthographic view showed that the wireframe has a bar going diagonally too. Clicking this reliably selects the CSGBox.

Also, despite only one diagonal bar being visible, there **is** a second diagonal bar, going from the top-right to the bottom-left corner. It's able to be clicked still, but it's invisible on the top side, being blocked by the mesh itself. 

<img width="852" height="858" alt="Image" src="https://github.com/user-attachments/assets/c053cbab-140c-4805-b105-7a269a03a1fe" />

Looking further, I noticed that the diagonal bar seems to work differently when dealing with overlapping boxes. In cases where a CSGBox is a parent to another CSGBox, clicking the diagonal bar on the child will almost always select the child, but clicking the child's outer wireframe seems to randomly choose whether to select the child, or it's parent.

(The left box is a parent to the right box.)

<img width="1200" height="582" alt="Image" src="https://github.com/user-attachments/assets/9d3cd321-69ae-49c5-abc4-7d7744b84932" />

<img width="1192" height="569" alt="Image" src="https://github.com/user-attachments/assets/72ab8e3a-73a0-4299-bb63-515409697c12" />

2. View Gizmo settings.

<img width="642" height="437" alt="Image" src="https://github.com/user-attachments/assets/2b5a2d2d-f37a-45f4-9845-c8007a114f6d" />

If you use the "View/Gizmos/CSGShape3D" to disable the wireframe, this doesn't affect the ability to **use** that gizmo. Clicking anywhere that the wireframe **should** be drawn, will still select the node as expected.

Apologies for the long comment, hopefully someone will find some use for these incoherent ramblings haha.
